test/liquidity-protection/LiquidityProtection.js

```diff
@@ -2325,6 +2338,277 @@ describe('LiquidityProtection', () => {
+                            expect(totalLiquidity).equal(RESERVE1_AMOUNT.add(depositAmount));
+                            expect(await systemPoolTokensBalance()).equals(poolTokenSupplyDelta.div(2));
+
+                            await liquidityProtection.setTotalPositionsValue(poolToken.address, totalLiquidity.mul(2));
+                            await liquidityProtection.migrateSystemPoolTokens([poolToken.address]);
+                            expect(await systemPoolTokensBalance()).equals(poolTokenSupplyDelta.div(2));
+                        });
+
+                        context('when the total positions value equals 80% of the total protected liquidity', async () => {
+                            it('should migrate 20% of the protected pool tokens', async () => {
+                                const provider = owner;
+                                await initPool(isEthReserve);
+
+                                const poolTokenSupply = await poolToken.totalSupply();
+                                await addV2Liquidity(provider, depositAmount, isEthReserve);
+                                const newPoolTokenSupply = await poolToken.totalSupply();
+                                const poolTokenSupplyDelta = newPoolTokenSupply.sub(poolTokenSupply)
+                                expect(await reserveTokenBalance()).equal(RESERVE1_AMOUNT.add(depositAmount));
+                                const systemBalance = await systemPoolTokensBalance();
+                                expect(systemBalance).equals(poolTokenSupplyDelta.div(2));
+
+                                const protectedPoolTokens = await poolToken.balanceOf(liquidityProtectionWallet.address);
+                                await liquidityProtection.setTotalPositionsValue(poolToken.address, depositAmount.mul(80).div(100));
+                                await liquidityProtection.migrateSystemPoolTokens([poolToken.address]);
+
+                                const newProtectedPoolTokens = await poolToken.balanceOf(liquidityProtectionWallet.address);
+                                expect(newProtectedPoolTokens).equals(protectedPoolTokens.mul(80).div(100));
+
+                                // 40% of the system pool token balance should be migrated
+                                expect(await systemPoolTokensBalance()).equals(systemBalance.mul(60).div(100));
+
+                                // another call to migrate should do nothing
+                                await liquidityProtection.migrateSystemPoolTokens([poolToken.address]);
+                                const newProtectedPoolTokens2 = await poolToken.balanceOf(liquidityProtectionWallet.address);
+                                expect(newProtectedPoolTokens2).equals(protectedPoolTokens.mul(80).div(100));                                
+                                expect(await systemPoolTokensBalance()).equals(systemBalance.mul(60).div(100));
```

Comment:
So where exactly do you verify `'should migrate 20% of the protected pool tokens'`?

Reply:
`expect(newProtectedPoolTokens).equals(protectedPoolTokens.mul(80).div(100));`
the system is left with 80% of what it had

Reply:
In that case, I'd change the title

Reply:
migrating 20% of the pool tokens leaves the protocol with 80% which is what we're testing, so I don't see an issue

Reply:
migrating 20% of the pool tokens leaves the protocol with 80% which is what we're testing, so I don't see an issue